Index: ndppd-0.2.6/src/proxy.cc
===================================================================
--- ndppd-0.2.6.orig/src/proxy.cc
+++ ndppd-0.2.6/src/proxy.cc
@@ -182,7 +182,7 @@ void proxy::handle_stateless_advert(cons
     ptr<session> se = find_or_create_session(taddr);
     if (!se) return;
     
-    if (_autowire == true && se->status() == session::WAITING) {
+    if (_autowire == true && (se->status() == session::WAITING || se->status() == session::INVALID)) {
         se->handle_auto_wire(saddr, ifname, use_via);
     }
 }
Index: ndppd-0.2.6/src/session.cc
===================================================================
--- ndppd-0.2.6.orig/src/session.cc
+++ ndppd-0.2.6/src/session.cc
@@ -286,7 +286,7 @@ void session::handle_auto_unwire(const s
 
 void session::handle_advert(const address& saddr, const std::string& ifname, bool use_via)
 {
-    if (_autowire == true && _status == WAITING) {
+    if (_autowire == true && (_status == WAITING || _status == INVALID)) {
         handle_auto_wire(saddr, ifname, use_via);
     }
     
